src/main/java/springboot/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package springboot.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import springboot.dto.cartitem.AddCartItemRequestDto;
+import springboot.dto.cartitem.CartItemDto;
+import springboot.dto.shoppingcart.ShoppingCartDto;
+import springboot.exeptions.EntityNotFoundException;
+import springboot.mapper.CartItemMapper;
+import springboot.mapper.ShoppingCartMapper;
+import springboot.model.CartItem;
+import springboot.repository.CartItemRepository;
+import springboot.repository.ShoppingCartRepository;
+import springboot.service.ShoppingCartService;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+
+    public ShoppingCartDto getCartById(Long userId) {
+        return shoppingCartRepository.findById(userId)
```

Comment:
Are you sure that the user and the cart will have the same ID in your realization? If this is not the case, the user will receive someone else's cart.